src/main/java/step3/application/Engineer.java

```diff
@@ -0,0 +1,30 @@
+package step3.application;
+
+import step3.domain.car.Car;
+import step3.domain.car.Cars;
+import step3.domain.car.Location;
+import step3.domain.power.Power;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Collections.unmodifiableList;
+import static step3.utils.ValidationUtils.checkArgument;
+
+public class Engineer {
+
+    private static final int DEFAULT_LOCATION = 0;
+
+    public static Cars createCars(Integer carCount, Power power) {
+        checkArgument(carCount != null, "carCount is Required");
+        checkArgument(power != null, "power is Required");
+
+        List<Car> cars = new ArrayList<>();
+        for (int i = 0; i < carCount; i++) {
+            Car car = new Car(Location.placeOn(DEFAULT_LOCATION));
+            cars.add(car);
+        }
+        return new Cars(unmodifiableList(cars), power);
+    }
```

Comment:
```suggestion
    public static Cars createCars(Integer carCount, Power power) {
        validRequiredValues(carCount, power);

        List<Car> cars = new ArrayList<>();
        for (int i = 0; i < carCount; i++) {
            Car car = new Car(Location.placeOn(DEFAULT_LOCATION));
            cars.add(car);
        }
        return new Cars(unmodifiableList(cars), power);
    }

    private static void validRequiredValues(Integer carCount, Power power) {
        checkArgument(carCount != null, "carCount is Required");
        checkArgument(power != null, "power is Required");
    }

```
이런 형태로 유효성 검증하는 부문을 메소드를 이용해 한번 더 감싸면 어떨까요?
어떤 장점과 단점이 있는지 한번 고민해보아도 좋을 것 같아요. :)

하나 더 이야기하자면 여기서 static을 사용해야 할 이유가 없다면 static 제거해도 좋을 것 같아요.

Reply:
> 어떤 장점과 단점이 있는지 한번 고민해보아도 좋을 것 같아요. :)

용대님이 제안해주신 방법의
장점은 다음과 같을 것 같아요.
- 파라미터 개수가 늘어나거나 null 체크 이외에 검증이 필요하더라도 createCars 메소드에서는 라인수가 늘어나지 않는다.(실제 검증은 validRequiredValues 메소드에서 진행하기때문에) 
- 검증로직을 제외한 주요 로직이 눈에 더 잘 들어온다.

단점이라면 어떤 검증을 하는지 확인하기 위해 validRequiredValues 메소드 구현코드를 확인해야 하는게 있을 것 같아요.

지금은 검증해야할게 별로 없어서 createCars메소드에서 직접 수행했는데
제안해주신 방법도 좋을 것 같아요 👍 

> 하나 더 이야기하자면 여기서 static을 사용해야 할 이유가 없다면 static 제거해도 좋을 것 같아요.

현재 Engineer 클래스에 인스턴수 변수가 없어서 굳이 Engineer 인스턴스를 생성하지 않기 위해 
static 메소드를 사용했는데 static 을 제거하는걸 권장해주신 이유를 알려주실 수 있나요?

`다시 생각해보니 Power를 인스턴스 변수로 만드는게 더 좋을 것 같다고 생각이 들어서 이 경우에는 static을 제거 해야겠지만 현재 코드의 경우에도 static을 제거 하는게 좋을까요??`

Reply:
static 제거를 이야기 드린 이유는 비동기로 실행되는 상황이나 멀티 스레드 환경에서
static을 사용한 코드는 반드시 문제를 야기시키기 때문입니다.

또한 static을 사용하다보면 자연스레 static한 메소드들이 많아지고 이는 지속적으로
유지보수가 좋지 않은 코드를 작성하게 만들기도 하기 때문입니다.

지금 미션 과정에서는 복잡한 내용이 없고 멀티 스레드 환경에서 동작하지 않기 때문에 문제가 되는 부분은 없습니다. 😄

Reply:
그런 단점들도 있었네요..
감사합니다!